### ISSUE TITLE
Error: Element with key 'surv.bart' not found in DictionaryLearner! Did you mean 'surv.parametric' / 'surv.rpart'?

### ISSUE THREAD

**Issue by Kuroshiwo** [mlr-org/mlr3extralearners#346]
### Description

I wanted to train a BART model but it is not in the DictionaryLearner.

> lrn()
<DictionaryLearner> with 103 stored values
Keys: classif.abess, classif.AdaBoostM1, classif.bart, classif.C50, classif.cforest, classif.ctree,
  classif.debug, classif.earth, classif.featureless, classif.fnn, classif.gam, classif.gamboost,
  classif.gausspr, classif.gbm, classif.glmboost, classif.glmer, classif.IBk,
  classif.imbalanced_rfsrc, classif.J48, classif.JRip, classif.ksvm, classif.liblinear,
  classif.lightgbm, classif.LMT, classif.lssvm, classif.mob, classif.OneR, classif.PART,
  classif.priority_lasso, classif.randomForest, classif.rfsrc, classif.rpart, dens.hist, dens.kde,
  dens.kde_ks, dens.locfit, dens.logspline, dens.mixed, dens.nonpar, dens.pen, dens.plug,
  dens.spline, regr.abess, regr.bart, regr.cforest, regr.ctree, regr.cubist, regr.debug, regr.earth,
  regr.featureless, regr.fnn, regr.gam, regr.gamboost, regr.gausspr, regr.gbm, regr.glm,
  regr.glmboost, regr.IBk, regr.ksvm, regr.liblinear, regr.lightgbm, regr.lmer, regr.M5Rules,
  regr.mars, regr.mob, regr.priority_lasso, regr.randomForest, regr.rfsrc, regr.rpart, regr.rsm,
  regr.rvm, surv.akritas, surv.aorsf, surv.blackboost, surv.cforest, surv.coxboost, surv.coxph,
  surv.coxtime, surv.ctree, surv.cv_coxboost, surv.cv_glmnet, surv.deephit, surv.deepsurv,
  surv.dnnsurv, surv.flexible, surv.gamboost, surv.gbm, surv.glmboost, surv.glmnet, surv.kaplan,
  surv.loghaz, surv.mboost, surv.nelson, surv.obliqueRSF, surv.parametric, surv.pchazard,
  surv.penalized, surv.priority_lasso, surv.ranger, surv.rfsrc, surv.rpart, surv.svm, surv.xgboost

According to this link: https://mlr-org.com/learners.html
It is supposed to be available.

So, how do I train the BART model?

I tried this to install BART model
install_learners("surv.bart")

But got an error:
Error: Element with key 'surv.bart' not found in DictionaryLearner! Did you mean 'surv.parametric' / 'surv.rpart'?

I tried this code chunk but no success

> library(mlr3)
> library(mlr3proba)
> library(distr6)

Attaching package: ‘distr6’

The following object is masked from ‘package:stats’:

    qqplot

The following object is masked from ‘package:base’:

    truncate

> library(BART)
Loading required package: nlme
Loading required package: nnet
Loading required package: survival
> learner = mlr3::lrn("surv.bart")
Error: Element with key 'surv.bart' not found in DictionaryLearner! Did you mean 'surv.rpart'?
>

### Reproducible example

library(mlr3extralearners)
library(mlr3pipelines)
library(mlr3proba)
library(distr6)
library(BART)

> lrn()
<DictionaryLearner> with 103 stored values
Keys: classif.abess, classif.AdaBoostM1, classif.bart, classif.C50, classif.cforest, classif.ctree,
  classif.debug, classif.earth, classif.featureless, classif.fnn, classif.gam, classif.gamboost,
  classif.gausspr, classif.gbm, classif.glmboost, classif.glmer, classif.IBk,
  classif.imbalanced_rfsrc, classif.J48, classif.JRip, classif.ksvm, classif.liblinear,
  classif.lightgbm, classif.LMT, classif.lssvm, classif.mob, classif.OneR, classif.PART,
  classif.priority_lasso, classif.randomForest, classif.rfsrc, classif.rpart, dens.hist, dens.kde,
  dens.kde_ks, dens.locfit, dens.logspline, dens.mixed, dens.nonpar, dens.pen, dens.plug,
  dens.spline, regr.abess, regr.bart, regr.cforest, regr.ctree, regr.cubist, regr.debug, regr.earth,
  regr.featureless, regr.fnn, regr.gam, regr.gamboost, regr.gausspr, regr.gbm, regr.glm,
  regr.glmboost, regr.IBk, regr.ksvm, regr.liblinear, regr.lightgbm, regr.lmer, regr.M5Rules,
  regr.mars, regr.mob, regr.priority_lasso, regr.randomForest, regr.rfsrc, regr.rpart, regr.rsm,
  regr.rvm, surv.akritas, surv.aorsf, surv.blackboost, surv.cforest, surv.coxboost, surv.coxph,
  surv.coxtime, surv.ctree, surv.cv_coxboost, surv.cv_glmnet, surv.deephit, surv.deepsurv,
  surv.dnnsurv, surv.flexible, surv.gamboost, surv.gbm, surv.glmboost, surv.glmnet, surv.kaplan,
  surv.loghaz, surv.mboost, surv.nelson, surv.obliqueRSF, surv.parametric, surv.pchazard,
  surv.penalized, surv.priority_lasso, surv.ranger, surv.rfsrc, surv.rpart, surv.svm, surv.xgboost

I tried this to install BART model
install_learners("surv.bart")

But got an error:
Error: Element with key 'surv.bart' not found in DictionaryLearner! Did you mean 'surv.parametric' / 'surv.rpart'?

I tried this code chunk but no success

> library(mlr3)
> library(mlr3proba)
> library(distr6)

Attaching package: ‘distr6’

The following object is masked from ‘package:stats’:

    qqplot

The following object is masked from ‘package:base’:

    truncate

> library(BART)
Loading required package: nlme
Loading required package: nnet
Loading required package: survival
> learner = mlr3::lrn("surv.bart")
Error: Element with key 'surv.bart' not found in DictionaryLearner! Did you mean 'surv.rpart'?
>

> sessionInfo()
R version 4.3.3 (2024-02-29 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22621)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Denver
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] BART_2.9.6              survival_3.5-8          nnet_7.3-19             nlme_3.1-164           
 [5] distr6_1.8.4            mlr3proba_0.6.0-9000    mlr3_0.18.0             mlr3pipelines_0.5.0-2  
 [9] mlr3extralearners_0.7.1 librarian_1.8.1        

loaded via a namespace (and not attached):
 [1] Matrix_1.6-5         gtable_0.3.4         dplyr_1.1.4          compiler_4.3.3       param6_0.2.4        
 [6] crayon_1.5.2         tidyselect_1.2.1     Rcpp_1.0.12          parallel_4.3.3       splines_4.3.3       
[11] globals_0.16.3       scales_1.3.0         uuid_1.2-0           RhpcBLASctl_0.23-42  lattice_0.22-5      
[16] ggplot2_3.5.0        R6_2.5.1             generics_0.1.3       ooplah_0.2.0         palmerpenguins_0.1.1
[21] backports_1.4.1      checkmate_2.3.1      future_1.33.1        tibble_3.2.1         mlr3viz_0.8.0       
[26] set6_0.2.6           munsell_0.5.0        paradox_0.11.1       pillar_1.9.0         rlang_1.1.3         
[31] utf8_1.2.4           lgr_0.4.4            mlr3misc_0.14.0      cli_3.6.2            withr_3.0.0         
[36] magrittr_2.0.3       dictionar6_0.1.3     digest_0.6.35        grid_4.3.3           rstudioapi_0.15.0   
[41] lifecycle_1.0.4      vctrs_0.6.5          glue_1.7.0           data.table_1.15.2    listenv_0.9.1       
[46] codetools_0.2-19     fansi_1.0.6          parallelly_1.37.1    colorspace_2.1-0     pkgconfig_2.0.3     
[51] tools_4.3.3         
>

**Comment by sebffischer** [comment]
Please run  the code below and then try again.

```r
remotes::install_github("mlr-org/mlr3extralearners")
```

**Comment by Kuroshiwo** [comment]
Thank you very much @sebffischer . It worked.

Although I got another error in BART. I will chase that through BART.
